addon/core/handler.ts

```diff
@@ -81,23 +91,123 @@ export default class TableHandler {
+      const filterWithSameKey = acc.find((filter) => filter.key === v.key)
+
+      if (filterWithSameKey) {
+        filterWithSameKey.value = v.value;
+      } else {
+        acc.push(v);
+      }
+
+      return acc;
+    }, [...column.filters])
+
+    column.filters = _filters;
+
+    return this.tableManager.upsertColumns({ columns: this.columns }).then(({ columns }) => {
+      this.columns = columns;
+      this.currentPage = 1;
+      this.fetchRows();
+    });
   }
 
-  // @ts-ignore
-  applyOrder(column: Column, order: Order): Promise<any> {
-    throw new Error('NotImplemented');
+  /**
+   * Apply ordering to a column
+   *
+   * @param {Column} column — The column we want to order by
+   * @param {OrderDirection} direction — The direction we want to order the column in
+   * @returns {Promise<any>}
+   */
+  async applyOrder(column: Column, direction: OrderDirection): Promise<any> {
+    const orderedColumn = this.columns.find((column) => column.order);
+
+    if (orderedColumn) {
+      orderedColumn.order = undefined;
+    }
+
+    column.order = { key: column.definition.key, direction };
+
+    return this.tableManager.upsertColumns({ columns: this.columns }).then(({ columns }) => {
+      this.columns = columns;
+      this.currentPage = 1;
+      this.fetchRows();
+    });
   }
 
-  // @ts-ignore
-  resetColumns(): Promise<any> {
-    throw new Error('NotImplemented');
+  /**
+   * Reset columns' filters and order attributes
+   *
+   * @param {Column[]} columns — The columns we want to reset the state for.
+   * @returns {Promise<any>}
+   */
+  async resetColumns(columns: Column[]): Promise<any> {
+    columns.forEach((column) => {
+      column.filters = [];
+      column.order = undefined;
+    });
+
+    return this.tableManager.upsertColumns({ columns: this.columns }).then(({ columns }) => {
+      this.columns = columns;
+      this.currentPage = 1;
+      this.fetchRows();
+    });
   }
 
   // @ts-ignore
   onBottomReached(): void {
     throw new Error('NotImplemented');
   }
+
+  /**
+   * Toggles an instance of Tether for a given column.
+   *
+   * @param {string} on - The column we want to create a Tether instance for
+   * @param {Tether.ITetherOptions} options — Options that should be used for the Tether instance.
+   */
+  triggerTetherContainer(on: string, options: Tether.ITetherOptions): void {
+    if (this.tetherOn !== on) {
+      this.tetherOn = on;
+
+      scheduleOnce('afterRender', this, () => {
+        if (this.tetherInstance) {
+          this.tetherInstance.setOptions(options);
+        } else {
+          this.tetherInstance = new Tether(options);
+        }
+
+        scheduleOnce('afterRender', this, () => {
+          // @ts-ignore
+          this.tetherInstance.element.classList.add(`js--visible`);
+        });
+      });
+    } else {
+      this.tetherOn = undefined;
+
+      if (this.tetherInstance) {
+        this.destroyTetherInstance();
+      }
```

Comment:
```suggestion
      this.destroyTetherInstance();
```
the check is already done in the `destroyTetherInstance` method